tests/unit_tests.py

```diff
@@ -292,6 +292,7 @@ def test_competition_a_list(self):
         try:
             competitions = api.competitions_list()
             self.assertGreater(len(competitions), 0)  # Assuming there should be some competitions
+            api.competitions_list_cli()
```

Comment:
The `_cli` methods do the I/O for the user. They have no return value. If they work, the test passes. The point of calling them is to ensure the new generated objects (from `kagglesdk`) do not break the existing code. I could wrap it in an exception handler, but I don't think that's necessary.

Reply:
> If they work, the test passes.

IMO the problem is that "work" really hasn't been verified.  The only thing the test knows is that the server responded with something that can be parsed.  But "something" could be the empty list, so it's not even necessarily checking that the objects themselves can be parsed (which I think is your intention).  The test really needs to verify that at least one competition was returned from the server.  This could be done by refactoring so there's a function that returns the parsed competition list or just by intercepting std out.

Reply:
It sounds like you want the result object to be exercised in the same way it would if the `_cli` method were called, without calling the `_cli` method. I'm lazy, and didn't want to do all that coding. But I guess I can.

Reply:
> The test really needs to verify that at least one competition was returned from the server.

That's done on line 294. Similarly, other tests should have asserts to verify functionality prior to `_cli` being invoked.

Reply:
Tests have been updated to show the verification w/o calling the `_cli` version.